mentoring/mrq.py

```diff
@@ -0,0 +1,161 @@
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+        default=[],
+    )
+    ignored_choices = List(
+        display_name=_("Ignored Choices"),
+        help=_(
+            "Specify the value[s] that are neither correct nor incorrect. "
+            "Any values not listed as required or ignored will be considered wrong."
+        ),
+        scope=Scope.content,
+        list_values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+        list_style='set',  # Underered, unique items. Affects the UI editor.
+        default=[],
+    )
+    hide_results = Boolean(display_name="Hide results", scope=Scope.content, default=False)
+    editable_fields = ('question', 'required_choices', 'ignored_choices', 'message', 'weight', 'hide_results', )
+
+    def describe_choice_correctness(self, choice_value):
+        if choice_value in self.required_choices:
+            return self._(u"Required")
+        elif choice_value in self.ignored_choices:
+            return self._(u"Ignored")
+        return self._(u"Not Acceptable")
+
+    def submit(self, submissions):
+        log.debug(u'Received MRQ submissions: "%s"', submissions)
+
+        score = 0
+
+        results = []
+        for choice in self.custom_choices:
+            choice_completed = True
+            choice_tips_html = []
+            choice_selected = choice.value in submissions
+            if choice.value in self.required_choices:
+                if not choice_selected:
+                    choice_completed = False
+            elif choice_selected and choice.value not in self.ignored_choices:
+                choice_completed = False
+            for tip in self.get_tips():
+                if choice.value in tip.values:
+                    choice_tips_html.append(tip.render('mentoring_view').content)
+
+            if choice_completed:
+                score += 1
+
+            choice_result = {
+                'value': choice.value,
+                'selected': choice_selected,
+            }
+            # Only include tips/results in returned response if we want to display them
+            if not self.hide_results:
+                loader = ResourceLoader(__name__)
+                choice_result['completed'] = choice_completed
+                choice_result['tips'] = loader.render_template('templates/html/tip_choice_group.html', {
+                    'tips_html': choice_tips_html,
+                })
+
+            results.append(choice_result)
+
+        self.student_choices = submissions
+
+        status = 'incorrect' if score <= 0 else 'correct' if score >= len(results) else 'partial'
+
+        result = {
+            'submissions': submissions,
+            'status': status,
+            'choices': results,
+            'message': self.message,
+            'weight': self.weight,
+            'score': float(score) / len(results),
```

Comment:
I just checked and the mentoring block doesn't allow the user to submit unless at least one choice is selected, so this shouldn't be possible. Nevertheless, I'll add in a guard to avoid dividing by zero in case anything else is changed in the future.